### ISSUE TITLE
TypeError: connection._createPacket is not a function (SharedStream)

### ISSUE THREAD

**Issue by Exhabition** [abalabahaha/eris#1268]
I encountered this issue a while ago, so I thought I would make an issue on GitHub so it can be tracked more easily

Discord conversation in the Eris server regarding this issue: <br>https://canary.discord.com/channels/831967755447828491/831974405701500931/844248226634268672

When looping over all connections in the SharedStream, `_createPacket` is used. This results in an error since in the VoiceConnection class, there no longer is a `_createPacket` function.

Related files:
SharedStream: https://github.com/abalabahaha/eris/blob/master/lib/voice/SharedStream.js
VoiceConnection: https://github.com/abalabahaha/eris/blob/master/lib/voice/VoiceConnection.js

**Comment by JustCat80** [comment]
Looking at dev, this may be fixed? 
I see no reference to `_createPacket` and `_sendPacket` exists now
https://github.com/abalabahaha/eris/blob/53da0d5111cb884bffd2f2c1d0d5674a98558c26/lib/voice/VoiceConnection.js#L821

https://github.com/abalabahaha/eris/blob/53da0d5111cb884bffd2f2c1d0d5674a98558c26/lib/voice/SharedStream.js#L233

**Comment by Exhabition** [comment]
> Looking at dev, this may be fixed?
> I see no reference to `_createPacket` and `_sendPacket` exists now
> https://github.com/abalabahaha/eris/blob/53da0d5111cb884bffd2f2c1d0d5674a98558c26/lib/voice/VoiceConnection.js#L821
> 
> https://github.com/abalabahaha/eris/blob/53da0d5111cb884bffd2f2c1d0d5674a98558c26/lib/voice/SharedStream.js#L233

What do you mean? `_createPacket` is referenced in the `_sendPacket` function?
Unless I am missing something, this is not fixed on the dev branch. 
Pretty much still identical to the master branch aside from some doc fixes

I have a fork on my profile that sorta shows what should be done:
https://github.com/Exhabition/eris/blob/75585898e7264443377c0f3ac13ef207a710316b/lib/voice/SharedStream.js#L233

**Comment by JustCat80** [comment]
What line on dev is _createPacket called? I can't find it

**Comment by Exhabition** [comment]
> What line on dev is _createPacket called? I can't find it

**Inside** `_sendPacket`, so line 233

**Comment by JustCat80** [comment]
it's not referenced within sendPacket tho? the exact code is

`    _sendPacket(packet) {
        if(this.udpSocket) {
            try {
                this.udpSocket.send(packet, 0, packet.length, this.udpPort, this.udpIP);
            } catch(e) {
                this.emit("error", e);
            }
        }
    }`

**Comment by Exhabition** [comment]
> it's not referenced within sendPacket tho? the exact code is
> 
> ` _sendPacket(packet) { if(this.udpSocket) { try { this.udpSocket.send(packet, 0, packet.length, this.udpPort, this.udpIP); } catch(e) { this.emit("error", e); } } }`

```js
connection._sendPacket(connection._createPacket(this.current.buffer)); 
```

Sorry if my wording was confusing, I meant that it was used to create a packet and then passed along in the function

**Comment by JustCat80** [comment]
Ah my bad I misunderstood, I see, since you already have a branch with the changes you should make a pull request to dev to fix it

**Comment by abalabahaha** [comment]
Fixed in 0.16.0
